File: services/cms-svc/src/main/java/com/supportme/pay/cmssvc/shopitem/ShopItemRepository.java
package com.supportme.pay.cmssvc.shopitem;

import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface ShopItemRepository extends JpaRepository<ShopItem, Long> {
    Optional<ShopItem> findByOrganizationIdAndSlug(Long organizationId, String slug);

    List<ShopItem> findByOrganizationId(Long organizationId);

    List<ShopItem> findByOrganizationIdAndActiveTrue(Long organizationId);
}

File: pay-backend/storefront-domain/src/main/kotlin/com/supportme/pay/storefront/domain/repository/ShopItemRepository.kt
package com.supportme.pay.storefront.domain.repository

import com.supportme.pay.storefront.domain.entity.ShopItem
import com.supportme.pay.storefront.domain.entity.User
import org.springframework.data.jpa.repository.JpaRepository

interface ShopItemRepository : JpaRepository<ShopItem, Long> {
    /** Odpowiednik `ShopItem::forUser($userId)->ordered()`. */
    fun findAllByOwnerOrderBySortAscIdAsc(owner: User): List<ShopItem>

    fun findByOwnerAndId(owner: User, id: Long): ShopItem?

    fun findByTagUid(tagUid: String): ShopItem?

    fun findByTagUidAndActiveTrue(tagUid: String): ShopItem?
}
